topology/probes/libvirt/golibvirt.go

```diff
@@ -0,0 +1,118 @@
+// +build !libvirt,linux
+
+/*
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package libvirt
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	golibvirt "github.com/digitalocean/go-libvirt"
+)
+
+type golibvirtDomain struct {
+	*golibvirt.Libvirt
+	golibvirt.Domain
+}
+
+func (d golibvirtDomain) GetName() (string, error) {
+	return d.Name, nil
+}
+
+func (d golibvirtDomain) GetXML() ([]byte, error) {
+	return d.XML(d.Name, 0)
+}
+
+func (d golibvirtDomain) GetState() (DomainState, int, error) {
+	state, err := d.DomainState(d.Name)
+	return state, 0, err
+}
+
+type golibvirtMonitor struct {
+	libvirt *golibvirt.Libvirt
+	conn    net.Conn
+}
+
+func (m *golibvirtMonitor) AllDomains() ([]domain, error) {
+	domains, err := m.libvirt.Domains()
+	if err != nil {
+		return nil, err
+	}
+	allDomains := make([]domain, len(domains))
+	for i, domain := range domains {
+		allDomains[i] = &golibvirtDomain{Libvirt: m.libvirt, Domain: domain}
+	}
+	return allDomains, nil
+}
+
+func (m *golibvirtMonitor) Stop() {
+	m.libvirt.Disconnect()
+}
+
+func newMonitor(ctx context.Context, probe *Probe) (*golibvirtMonitor, error) {
+	if probe.uri == "" {
+		probe.uri = "unix:///var/run/libvirt/libvirt-sock"
```

Comment:
You probably want to make this uri configurable as location may vary on different systems: https://www.redhat.com/archives/libvirt-users/2012-October/msg00137.html

Reply:
`uri` comes from the configuration file, so it's configurable